### DISCUSSION TITLE
A few minor issues

### DISCUSSION THREAD

**Discussion by louispotok** [kuzmoyev/google-calendar-simple-api#75] [General]
Hi, it's my first time using gcsa -- super easy to get started, thanks so much for your work on this!

I encountered a few problems in trying to use it, but it didn't seem very friendly to jump into the issue tracker and dump several new issues at once. So I'm putting them here -- happy to add issues to the tracker if that's helpful, or to leave the discussion here for now.

I'm trying to do something very simple:

```
import datetime
from gcsa.google_calendar import GoogleCalendar

calendar = GoogleCalendar('myemail')
events = calendar.get_events(time_min=datetime.date(2021,1,1))
all_events = [e for e in events]
```

1. `Event.__init__` assumes that events have either a start or end date, but it's possible for them to have neither. For example, I got an event returned that looked like this: ```{'kind': 'calendar#event', 'etag': '"redacted"', 'status': 'cancelled', 'originalStartTime': {'dateTime': '2021-01-01T15:00:00-05:00', 'timeZone': 'America/Los_Angeles'}}```. This caused an attribute error on line 131 of `event.py`. I "fixed" it by adding an `else` to the block above where `self.end` is set, making it None. Not sure if this causes downstream problems. 

2. Took me a bit of poking around to find that the `transparency` field is included under `Event.other` when available. Doesn't look like the `other` field is covered in the docs -- maybe add a mention of that, and even put in a list of other fields so that it's searchable.

3. Some of my events have Attachments of unsupported types, which causes the whole event serialization to error. I'm not sure where attachments are used elsewhere, but a simple workaround might be:
* allowing Attachments to be created with unsupported mime types
* adding a boolean field on the Attachment object for `unsupported_mime_type` which can be checked later.

**Comment by kuzmoyev** [comment]
Hi @louispotok,

Thanks for the information. All points are good. Feel free to submit issues for all of them. Or I'll move them to the issues in a free time.

Unfortunately, Google's docs are not very strict and complete about the formats, so still tuning the `gcsa` with the help of users like you. So thank you for your help. I will address these issues in the next update.

**Comment by kuzmoyev** [comment]
Also, If you have fixes for any of the issues, PRs are very welcomed :)

**Reply by louispotok** [reply]
Submitted for (1) and (3), and created #78 to track (2) -- happy to take that on as well. Thanks for the quick response!

**Reply by kuzmoyev** [reply]
Thank you for the PRs and issues!

I've added a couple of comments there for the minor changes.
